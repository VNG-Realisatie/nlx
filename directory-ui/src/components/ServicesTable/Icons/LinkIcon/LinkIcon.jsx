import React from 'react'
import { oneOf } from 'prop-types'
import StyledLinkIcon from './LinkIcon.styles'

const colors = {
  blue: '#517FFF',
  grey: '#CAD0E0'
}

const LinkIcon = ({ color }) =>
  <StyledLinkIcon viewBox="0 0 24 24">
    <g fill="none" fillRule="evenodd">
      <circle cx="12" cy="12" r="12"/>
      <path
        d="M6.845 6.841a2.792 2.792 0 0 1 3.946 0l2.545 2.546 1.21-1.21L12 5.633a4.502 4.502 0 0 0-6.364 0 4.502 4.502 0 0 0 0 6.364l2.546 2.546 1.209-1.21-2.546-2.545a2.792 2.792 0 0 1 0-3.946zm1.973 3.246l5.091 5.09 1.273-1.272-5.091-5.091-1.273 1.273zm9.546 1.909L15.818 9.45l-1.209 1.21 2.546 2.545a2.792 2.792 0 0 1 0 3.946 2.792 2.792 0 0 1-3.946 0l-2.545-2.546-1.21 1.21L12 18.36a4.502 4.502 0 0 0 6.364 0 4.502 4.502 0 0 0 0-6.364z"
        fill={colors[color]} fillRule="nonzero"/>
    </g>
  </StyledLinkIcon>

LinkIcon.propTypes = {
  color: oneOf(['blue', 'grey']).isRequired
}

export default LinkIcon
