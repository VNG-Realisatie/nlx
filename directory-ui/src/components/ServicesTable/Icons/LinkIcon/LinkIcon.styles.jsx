import styled from 'styled-components'

export default styled.svg`
  display: inline-block
  width: 24px
  height: 24px
`