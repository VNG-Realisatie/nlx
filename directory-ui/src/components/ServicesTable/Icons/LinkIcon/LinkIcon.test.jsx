import React from 'react'
import { shallow } from 'enzyme'
import LinkIcon from '../LinkIcon/LinkIcon';

it('should support the grey color', () => {
  expect(shallow(<LinkIcon color="grey"/>)).toMatchSnapshot()
})

it('should support the blue color', () => {
  expect(shallow(<LinkIcon color="blue"/>)).toMatchSnapshot()
})
