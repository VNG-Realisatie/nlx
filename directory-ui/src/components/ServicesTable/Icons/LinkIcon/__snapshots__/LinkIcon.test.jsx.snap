// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should support the blue color 1`] = `
<styled.svg
  viewBox="0 0 24 24"
>
  <g
    fill="none"
    fillRule="evenodd"
  >
    <circle
      cx="12"
      cy="12"
      r="12"
    />
    <path
      d="M6.845 6.841a2.792 2.792 0 0 1 3.946 0l2.545 2.546 1.21-1.21L12 5.633a4.502 4.502 0 0 0-6.364 0 4.502 4.502 0 0 0 0 6.364l2.546 2.546 1.209-1.21-2.546-2.545a2.792 2.792 0 0 1 0-3.946zm1.973 3.246l5.091 5.09 1.273-1.272-5.091-5.091-1.273 1.273zm9.546 1.909L15.818 9.45l-1.209 1.21 2.546 2.545a2.792 2.792 0 0 1 0 3.946 2.792 2.792 0 0 1-3.946 0l-2.545-2.546-1.21 1.21L12 18.36a4.502 4.502 0 0 0 6.364 0 4.502 4.502 0 0 0 0-6.364z"
      fill="#517FFF"
      fillRule="nonzero"
    />
  </g>
</styled.svg>
`;

exports[`should support the grey color 1`] = `
<styled.svg
  viewBox="0 0 24 24"
>
  <g
    fill="none"
    fillRule="evenodd"
  >
    <circle
      cx="12"
      cy="12"
      r="12"
    />
    <path
      d="M6.845 6.841a2.792 2.792 0 0 1 3.946 0l2.545 2.546 1.21-1.21L12 5.633a4.502 4.502 0 0 0-6.364 0 4.502 4.502 0 0 0 0 6.364l2.546 2.546 1.209-1.21-2.546-2.545a2.792 2.792 0 0 1 0-3.946zm1.973 3.246l5.091 5.09 1.273-1.272-5.091-5.091-1.273 1.273zm9.546 1.909L15.818 9.45l-1.209 1.21 2.546 2.545a2.792 2.792 0 0 1 0 3.946 2.792 2.792 0 0 1-3.946 0l-2.545-2.546-1.21 1.21L12 18.36a4.502 4.502 0 0 0 6.364 0 4.502 4.502 0 0 0 0-6.364z"
      fill="#CAD0E0"
      fillRule="nonzero"
    />
  </g>
</styled.svg>
`;
